cmd/controller/main.go

```diff
@@ -355,9 +362,15 @@ func parseEnvFlags() config {
 		logger.WithError(err).Fatalf("could not parse %s", sidecarMemoryLimitFlag)
 	}
 
+	portRanges, err := parsePortRanges(viper.GetString(additionalPortRangesFlag))
```

Comment:
It's a little awkward here that you're carrying the default port range and the additional port ranges separately. I'm ok with them split on the config surface for backwards compat, but in the controller, my thinking is that we should just add `default` to the map right here and change `config` to use the `portRanges` map.